pkg/utils/common/finalizer.go

```diff
@@ -125,3 +127,27 @@ func EnsureDeleteServiceFinalizer(service *corev1.Service, key string, kubeClien
 	svcLogger.V(2).Info("Removing finalizer from service", "finalizerKey", key)
 	return patch.PatchServiceObjectMetadata(kubeClient.CoreV1(), service, *updatedObjectMeta)
 }
+
+// EnsureServiceDeleteFinalizers patches the service to ensure the specified finalizers are not present in the service finalizers list.
+// This function is needed if more than one finalizer has to be removed since you can't invoke the 1 param version multiple times.
+// This function will not modify the service object passed as the argument. Instead, a deep copy will be used to do a patch.
+func EnsureServiceDeleteFinalizers(service *corev1.Service, ensureRemoveKeys []string, kubeClient kubernetes.Interface, svcLogger klog.Logger) error {
+	var needToRemove []string
+	for _, key := range ensureRemoveKeys {
+		if HasGivenFinalizer(service.ObjectMeta, key) {
+			needToRemove = append(needToRemove, key)
+		}
+	}
+	if len(needToRemove) == 0 {
+		return nil
+	}
+
+	// Make a copy of object metadata so we don't mutate the shared informer cache.
+	updatedObjectMeta := service.ObjectMeta.DeepCopy()
+	for _, key := range needToRemove {
+		updatedObjectMeta.Finalizers = slice.RemoveString(updatedObjectMeta.Finalizers, key, nil)
+	}
+
+	svcLogger.V(2).Info("Removing finalizers from service", "finalizerKeys", ensureRemoveKeys)
```

Comment:
Replace ensureRemoveKeys with needToRemove

Reply:
Instead of having 2 for loops you can combine both in one:
```suggestion
	updatedObjectMeta := service.ObjectMeta.DeepCopy()
	modified := false 

	for _, key := range ensureRemoveKeys {
		if HasGivenFinalizer(service.ObjectMeta, key) {
			updatedObjectMeta.Finalizers = slice.RemoveString(updatedObjectMeta.Finalizers, key, nil)
			svcLogger.V(2).Info("Removing finalizer from service", "finalizerKey", key)
			modified = true
		}
	}

	if !modified {
		return nil
	}

	return patch.PatchServiceObjectMetadata(kubeClient.CoreV1(), service, *updatedObjectMeta)
```

Reply:
this was actually part of the parent commit that was merged yesterday